tools/pycafe/create_pycafe_links.py

```diff
@@ -0,0 +1,131 @@
+"""Generate PyCafe links for the example dashboards and post them as a comment on the pull request and as status."""
+
+import base64
+import datetime
+import gzip
+import json
+import os
+import subprocess
+import sys
+import textwrap
+from pathlib import Path
+from typing import Optional
+from urllib.parse import quote, urlencode
+
+from github import Auth, Github
+
+GITHUB_TOKEN = str(os.getenv("GITHUB_TOKEN"))
+REPO_NAME = str(os.getenv("GITHUB_REPOSITORY"))
+PR_NUMBER = int(os.getenv("PR_NUMBER"))
+RUN_ID = str(os.getenv("RUN_ID"))
+PACKAGE_VERSION = subprocess.check_output(["hatch", "version"]).decode("utf-8").strip()
+PYCAFE_URL = "https://py.cafe"
+
+# Access
+auth = Auth.Token(GITHUB_TOKEN)
+g = Github(auth=auth)
+
+# Get PR and commits
+repo = g.get_repo(REPO_NAME)
+pr = repo.get_pull(PR_NUMBER)
+commit_sha = pr.head.sha
+commit = repo.get_commit(commit_sha)
+
+
+def generate_link(directory: str, extra_requirements: Optional[list[str]] = None):
+    """Generate a PyCafe link for the example dashboards."""
+    base_url = f"https://raw.githubusercontent.com/mckinsey/vizro/{commit_sha}/vizro-core/{directory}"
```

Comment:
I feel like this should (or at least the first part of it which isn't templated) should be extracted to a constant at the top if `PYCAFE_URL` is there.

Or just hardcode in `PYCAFE_URL` and remove the constant.